Sources/System/FileHelpers.swift

```diff
@@ -79,6 +79,75 @@ extension FileDescriptor {
     }
   }
 
+  /// Reads bytes at the current file offset into a buffer until the buffer is filled.
+  ///
+  /// - Parameters:
+  ///   - buffer: The region of memory to read into.
+  /// - Returns: The number of bytes that were read, equal to `buffer.count`.
```

Comment:
If it's always equal, then we'd return a Bool. But, if we might only partly fill for an oversized buffer, then it makes sense to return the count.

Reply:
That makes sense. If we go the route of returning a `Bool` are we happy to accept the asymmetry it will create with the `writeAll(_:)` helpers, which currently return `Int` even though the docs for `writeAll(_:)` state:

```swift
  /// Writes a sequence of bytes to the current offset
  /// and then updates the offset.
  ///
  /// - Parameter sequence: The bytes to write.
  /// - Returns: The number of bytes written, equal to the number of elements in `sequence`.
  ///
  /// This method either writes the entire contents of `sequence`,
  /// or throws an error if only part of the content was written.
...
  @_alwaysEmitIntoClient
  @discardableResult
  public func writeAll<S: Sequence>(
...
```

Reply:
Beyond consistency, returning an integer has the very slight convenience benefit that I can write `c += read(filling: buffer)` to keep a running total of bytes read. (`read(filling: buffer); c += buffer.count` works but it is more error prone)

Returning a boolean instead of throwing an error on EOF is an option to consider. It's sort of consistent with the original interfaces, but it has the same convenience issue as returning a partial buffer -- callers would need to manually check for it and in the partial case they will typically end up throwing anyway.

So I'm slightly preferring having this return `buffer.count`.

Reply:
Current state in PR returns an `Int`, discussed in more detail in https://github.com/apple/swift-system/pull/84#discussion_r878251684.